apps/src/scratch/scratch.js

```diff
@@ -0,0 +1,193 @@
+import AudioEngine from 'scratch-audio';
+import Renderer from 'scratch-render';
+import Storage from 'scratch-storage';
+import VM from 'scratch-vm';
+import Blockly from 'scratch-blocks/dist/vertical.js';
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+
+import { getStore, registerReducers } from '@cdo/apps/redux';
+import * as commonReducers from '@cdo/apps/redux/commonReducers';
+import { singleton as studioApp } from '@cdo/apps/StudioApp';
+import ScratchView from './ScratchView';
+import { scratchDefaultProject } from './scratchDefaultProject';
+
+const Scratch = window.Scratch = window.Scratch || {};
+
+export default function init(options) {
+  registerReducers(commonReducers);
+
+  options.maxVisualizationWidth = 480;
+  options.vizAspectRatio = 4 / 3;
+  options.enableShowCode = false;
+  options.pinWorkspaceToBottom = true;
+  options.skin = {};
+  window.appOptions = options;
+
+  studioApp().configure(options);
+  studioApp().setPageConstants(options, {});
+
+  ReactDOM.render(
+    <Provider store={getStore()}>
+      <ScratchView onMount={onMount} />
+    </Provider>,
+    document.getElementById(options.containerId),
+  );
+
+  function onMount() {
+    studioApp().init(options);
+  }
+
+  /**
+   * @param {Asset} asset - calculate a URL for this asset.
+   * @returns {string} a URL to download a project asset (PNG, WAV, etc.)
+   */
+  function getAssetUrl(asset) {
+    return studioApp().assetUrl(`media/scratch/${asset.assetId}.${asset.dataFormat}`);
+  }
+
+  // Instantiate the VM.
+  const vm = new VM();
+  Scratch.vm = vm;
+  options.getCode = vm.saveProjectSb3.bind(vm);
+
+  const storage = new Storage();
+  const AssetType = storage.AssetType;
+  storage.addWebSource([AssetType.ImageVector, AssetType.ImageBitmap, AssetType.Sound], getAssetUrl);
+  vm.attachStorage(storage);
+
+  // Instantiate the renderer and connect it to the VM.
+  const canvas = document.getElementById('scratch-stage');
+
+  // PhantomJS doesn't support WebGL.
+  if (!IN_UNIT_TEST) {
+    const renderer = new Renderer(canvas);
+    Scratch.renderer = renderer;
+    vm.attachRenderer(renderer);
+    const audioEngine = new AudioEngine();
+    vm.attachAudioEngine(audioEngine);
+  }
```

Comment:
Definitely!  I'd like to get [Headless Chrome setup to replace PhantomJS](https://developers.google.com/web/updates/2017/06/headless-karma-mocha-chai).

PhantonJS has no WebGL support.  I tried mocking with https://github.com/kbirk/webgl-mock but https://github.com/greggman/twgl.js could tell the types weren't right and kept throwing.